### PR TITLE
[CHK-1549] feat: update refund logic

### PR DESCRIPTION
**Pull request by MarcoAGreco**
#### List of Changes

-  Avoid refund retry when `CONFLICT - 409` is returned as pgs response

#### Motivation and Context

- Avoid retry for unrecoverable refund error
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/exceptions/RefundNotAllowedException.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package it.pagopa.ecommerce.eventdispatcher.exceptions
+
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.ResponseStatus
+
+@ResponseStatus(value = HttpStatus.BAD_GATEWAY)
+class RefundNotAllowedException(detail: String) :
```
Comment (Fernando-Granato) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/exceptions/RefundNotAllowedException.kt:7]: Why do you pass a detail if it is not used ?

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt [hunk 2]
```diff
@@ -202,7 +205,13 @@ fun refundTransaction(
         } else {
           Mono.just(tx)
         }
-        .flatMap { refundRetryService.enqueueRetryEvent(it, retryCount) }
+        .flatMap {
+          when(exception){
+            // Enqueue retry event only if refund is allowed
+            !is RefundNotAllowedException -> refundRetryService.enqueueRetryEvent(it, retryCount)
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt:211]: add junit test to cover the case when retry event is enqueued and when is not based on the PGS received http response code

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/exceptions/RefundNotAllowedException.kt (hunk 1 above)
Comment (MarcoAGreco) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/exceptions/RefundNotAllowedException.kt]: Right, solved with [e319b38](https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/109/commits/e319b3809f5fc94abcbd7503b350a7bf95e4eaf7) :)

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt (hunk 2 above)
Comment (MarcoAGreco) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt:209]: You're right, done with [4db90e8](https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/109/commits/4db90e81add31093172d938690cb7875eebfa5e6) 👍

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**